test/test.js

```diff
@@ -131,8 +136,114 @@ describe('Service', function() {
       assert.notExists(properties.getProperty(key));
     });
   });
+
+  describe('#hasAccess()', function() {
+    it('should use the lock to prevent concurrent access', function(done) {
+      var token = {
+        granted_time: 100,
+        expires_in: 100,
+        refresh_token: 'bar'
+      };
+      var properties = new MockProperties();
+      properties.setProperty('oauth2.test', JSON.stringify(token));
+
+      mocks.UrlFetchApp.delayFunction = () => 100;
+      mocks.UrlFetchApp.resultFunction = () =>
+        JSON.stringify({
+          access_token: Math.random().toString(36)
+        });
```

Comment:
This may be a bit more readable:
```js
      mocks.UrlFetchApp.resultFunction = () => JSON.stringify({
        access_token: Math.random().toString(36)
      });
```
If you don't apply this, the indentation for statement continuation should be fixed.

Reply:
Done